### ISSUE TITLE
Add overflow mode when there are too many icons

### ISSUE THREAD

**Issue by muhammadyusuf-kurbonov** [ubuntu/gnome-shell-extension-appindicator#363]
Implement, please overflow mode when there are too many icons. The count of icons can be set via the extension's settings.
Example: https://github.com/MartinPL/Tray-Icons-Reloaded#overflow-mode

**Comment by 3v1n0** [comment]
Yeah, i've been thinking on this for long time, the behavior should be a bit different but I agree would be nice.

**Comment by AsciiWolf** [comment]
I personally think a similar behaviour that Windows have would be ideal: When there are too many tray icons (appindicators), display only a reasonable amount of them and hide the others behind an arrow (or similar) icon with a pop-up menu.

**Comment by 3nr8** [comment]
That would be really great but please let us decide which icons to put off. :-)

**Comment by muhammadyusuf-kurbonov** [comment]
Any news about this feature? Can we expect it in Q1 of 2023?

**Comment by Yasand123** [comment]
This would be really nice. Also please make it configurable, personally I want all the tray icons to be hidden in the overflow menu.

**Comment by redeven** [comment]
Adding my vote for this feature. Being able to hide tray icons into an overflow menu to keep things tidy would work wonders.

It could hide either specific icons, or N icons in order (and just order them with [another extension](https://extensions.gnome.org/extension/2114/order-gnome-shell-extensions/) to keep the desired ones visible).

**Comment by Impasse52** [comment]
Also adding my own vote for this feature. I think one of Gnome's main attractions is how clutterless and clean it is, and this kind of feature would maintain this aspect while still not missing out on any feature that would be lost by not using appindicators altogether. It'd be great to have a configurable overflow menu.

**Comment by NubsCarson** [comment]
Adding a concrete reproduction from GNOME Shell 49.5 / extension v64.

I originally hit this through Brave exposing many Chromium StatusNotifierItems (tracked at https://github.com/brave/brave-browser/issues/56896). That source bug aside, I tested the generic overflow behavior with a temporary fake SNI process so it does not depend on Brave.

Baseline watcher state:

```text
as 3 ":1.2626@/StatusNotifierItem" ":1.3791@/StatusNotifierItem" ":1.4927@/StatusNotifierItem"
```

Then I registered 20 fake `org.kde.StatusNotifierItem` objects from one D-Bus owner. The watcher accepted all of them and the extension rendered them all into the panel:

```text
as 23 ":1.2626@/StatusNotifierItem" ":1.3791@/StatusNotifierItem" ":1.4927@/StatusNotifierItem" ":1.8787@/FakeSNI/Item1" ":1.8787@/FakeSNI/Item2" ... ":1.8787@/FakeSNI/Item20"
```

After killing the fake process, the watcher returned to the baseline 3 items.

So this issue is reproducible as a host-side layout/UX problem independent of the original application bug: a valid SNI owner can register enough indicators that the right side of the GNOME panel becomes crowded and system controls can become hard to reach. An overflow/collapse affordance, or a configurable max visible indicator count with the rest behind an overflow button, would address that class of failure while still showing every item somewhere.
